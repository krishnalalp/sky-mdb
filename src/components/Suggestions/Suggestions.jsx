import React from "react";
import { Divider, List } from "antd";
import { useSelector } from "react-redux";
import PropTypes from "prop-types";
import { getSuggestions } from "../../app/reducer";

export function Suggestions({ handleClick }) {
  const suggestions = useSelector(getSuggestions);
  const { movies, actors, shows } = suggestions;

  const renderList = (title, type, arr) => {
    const newarr = arr.slice(0, 3);
    if (newarr.length) {
      return (
        <>
          <Divider orientation="left">{title}</Divider>
          <List
            bordered
            dataSource={newarr}
            renderItem={(item) => (
              <List.Item onClick={() => handleClick(`/${type}/${item.id}`)}>
                {item.title || item.name}
              </List.Item>
            )}
          />
        </>
      );
    }
    return <></>;
  };

  return (
    <div>
      <>
        {renderList("Movies", "movie", movies)}
        {renderList("Actors", "person", actors)}
        {renderList("Shows", "tv", shows)}
      </>
    </div>
  );
}

Suggestions.propTypes = {
  handleClick: PropTypes.func.isRequired,
};
