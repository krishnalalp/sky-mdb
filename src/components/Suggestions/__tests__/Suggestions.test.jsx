import React from "react";
import { useSelector } from "react-redux";
import renderer from "react-test-renderer";
import { Suggestions } from "../Suggestions";
import { suggestionsData } from "./mockData";

test("Suggestions component rendered", () => {
  useSelector.mockImplementation(() => suggestionsData);

  const component = renderer.create(<Suggestions handleClick={() => {}} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
