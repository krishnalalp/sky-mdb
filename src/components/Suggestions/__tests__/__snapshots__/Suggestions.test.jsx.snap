// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggestions component rendered 1`] = `
<div>
  <div
    className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-left"
    role="separator"
  >
    <span
      className="ant-divider-inner-text"
    >
      Movies
    </span>
  </div>
  <div
    className="ant-list ant-list-split ant-list-bordered"
  >
    <div
      className="ant-spin-nested-loading"
    >
      <div
        className="ant-spin-container"
      >
        <ul
          className="ant-list-items"
        >
          <li
            className="ant-list-item"
            onClick={[Function]}
          >
            Captain Marvel
          </li>
          <li
            className="ant-list-item"
            onClick={[Function]}
          >
            Marvel's Behind the Mask
          </li>
          <li
            className="ant-list-item"
            onClick={[Function]}
          >
            Captain Marvel 2
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-left"
    role="separator"
  >
    <span
      className="ant-divider-inner-text"
    >
      TV Shows
    </span>
  </div>
  <div
    className="ant-list ant-list-split ant-list-bordered"
  >
    <div
      className="ant-spin-nested-loading"
    >
      <div
        className="ant-spin-container"
      >
        <ul
          className="ant-list-items"
        >
          <li
            className="ant-list-item"
            onClick={[Function]}
          >
            Marvel Studios: Legends
          </li>
          <li
            className="ant-list-item"
            onClick={[Function]}
          >
            Marvel's Agents of S.H.I.E.L.D.
          </li>
          <li
            className="ant-list-item"
            onClick={[Function]}
          >
            Marvel's Daredevil
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-left"
    role="separator"
  >
    <span
      className="ant-divider-inner-text"
    >
      Cast
    </span>
  </div>
  <div
    className="ant-list ant-list-split ant-list-bordered"
  >
    <div
      className="ant-spin-nested-loading"
    >
      <div
        className="ant-spin-container"
      >
        <ul
          className="ant-list-items"
        >
          <li
            className="ant-list-item"
            onClick={[Function]}
          >
            Marve Feinberg
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
